File: mobile_app/lib/data/auth_repository.dart
import 'api_client.dart';
import 'models/auth_models.dart';

class AuthRepository {
  final ApiClient _apiClient;

  AuthRepository({ApiClient? apiClient}) : _apiClient = apiClient ?? ApiClient();

  Future<AuthResponse> login({required String email, required String password}) async {
    final data = await _apiClient.postJson('/auth/login', {
      'email': email,
      'password': password,
    });
    return AuthResponse.fromJson(data);
  }

  Future<AuthResponse> register({
    required String firstName,
    required String lastName,
    required String email,
    required String password,
  }) async {
    final data = await _apiClient.postJson('/auth/register', {
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'password': password,
    });
    return AuthResponse.fromJson(data);
  }
}

File: mobile_app/src/data/authRepository.ts
import { apiClient } from './apiClient';
import { parseAuthResponse } from './models';

export const authRepository = {
  async login(email: string, password: string) {
    const data = await apiClient.postJson('/auth/login', { email, password });
    return parseAuthResponse(data);
  },

  async register(input: {
    firstName: string;
    lastName: string;
    email: string;
    password: string;
  }) {
    const data = await apiClient.postJson('/auth/register', input);
    return parseAuthResponse(data);
  },
};
